### PR TITLE
(feat) O3-3112: Add form collapse toggle to the `esm-form-engine-app`

### PR DESCRIPTION
**Pull request by usamaidrsk**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This PR add the form collapse component to the form-engine-app as use it as an extension in the `patient-chart-workspace-header-slot`

## Screenshots

image

https://github.com/openmrs/openmrs-esm-patient-chart/assets/53287480/0b68ea45-68da-4595-b366-55de5154e9e8

## Related Issue

[O3-3112](https://openmrs.atlassian.net/browse/O3-3112)

## Other

[O3-3112]: https://openmrs.atlassian.net/browse/O3-3112?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pirupius** [comment]
@usamaidrsk please add a video/gif showing the behaviour

File: packages/esm-form-engine-app/src/index.ts [hunk 1]
```diff
@@ -16,3 +15,10 @@ export function startupApp() {
 }
 
 export const formRenderer = getAsyncLifecycle(() => import('./form-renderer/form-renderer.component'), options);
+export const FormCollapseToggle = getAsyncLifecycle(
```
Comment (pirupius) [inline packages/esm-form-engine-app/src/index.ts:18]:
```suggestion
export const formCollapseToggle = getAsyncLifecycle(
```

File: packages/esm-form-engine-app/src/form-collapse-toggle/form-collapse-toggle.component.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    };
+
+    window.addEventListener('openmrs:form-view-embedded', handleFormEmbedded);
+
+    return () => {
+      window.removeEventListener('openmrs:form-view-embedded', handleFormEmbedded);
+    };
+  }, []);
+
+  const handleOnToggle = (value: boolean) => {
```
Comment (ibacher) [inline packages/esm-form-engine-app/src/form-collapse-toggle/form-collapse-toggle.component.tsx:22]: This should be wrapped in `useCallback()` or hoisted out of the React component.

File: packages/esm-form-engine-app/src/index.ts [hunk 3]
```diff
@@ -16,3 +15,10 @@ export function startupApp() {
 }
 
 export const formRenderer = getAsyncLifecycle(() => import('./form-renderer/form-renderer.component'), options);
+export const formCollapseToggle = getAsyncLifecycle(
```
Comment (ibacher) [inline packages/esm-form-engine-app/src/index.ts:18]:
```suggestion

export const formCollapseToggle = getAsyncLifecycle(
```

**Review by ibacher — COMMENTED** [review]
LGTM. Thanks @usamaidrsk!

**Review by ibacher — APPROVED** [review]

**Pull request merged**